validator/sawtooth_validator/gossip/gossip.py

```diff
@@ -737,9 +744,12 @@ def _remove_temporary_connection(self, connection_id):
         if status == PeerStatus.TEMP:
             LOGGER.debug("Closing connection to %s", connection_id)
             msg = DisconnectMessage()
-            self._network.send(validator_pb2.Message.NETWORK_DISCONNECT,
-                               msg.SerializeToString(),
-                               connection_id)
+            try:
+                self._network.send(validator_pb2.Message.NETWORK_DISCONNECT,
+                                   msg.SerializeToString(),
+                                   connection_id)
+            except ValueError:
+                pass
```

Comment:
If we pass here, doesn't this imply that the connection has been removed and the next line will throw a key error?

Reply:
No, It only gets into this block if the connection_id is in self._connection_statuses[connection_id]  and needs to be removed. Connection_statues are gossips own copy of known connections. Also, network.remove_connection does not throw an error if the connection_id is already gone.